### ISSUE TITLE
Parameter node wrapper GUI connectors are missing for many widgets

### ISSUE THREAD

**Issue by lassoan** [Slicer/Slicer#7308]
## Summary

Many Qt widgets cannot be used with the recently added parameter node wrapper infrastructure.

## Steps to reproduce

Try using the "not supported" widgets in a module GUI and link to parameter node.

Supported:
- qt.QCheckBox
- qt.QPushButton
- qt.QSlider
- qt.QSpinBox
- qt.QDoubleSpinBox
- ctk.ctkSliderWidget
- slicer.qMRMLSliderWidget
- qt.QComboBox
- qt.QLineEdit
- qt.QTextEdit
- ctk.ctkRangeWidget
- ctk.ctkPathLineEdit
- ctk.ctkDirectoryButton
- slicer.qMRMLNodeComboBox

Not supported:
- qMRMLCheckableNodeComboBox
- qMRMLColorPickerWidget
- qMRMLCoordinatesWidget
- qMRMLLinearTransformSlider
- qMRMLListWidget
- qMRMLMatrixWidget
- qMRMLRangeWidget
- qMRMLSpinBox
- qMRMLTransformSliders
- ctkCheckableComboBox
- ctkCheckBox
- ctkColorPickerButton
- ctkComboBox
- ctkCoordinatesWidget
- ctkCoordinatesWidget
- ctkDoubleRangeSlider
- ctkDoubleSlider
- ctkDoubleSpinBox
- ctkFittedTextBrowser
- ctkMatrixWidget
- ctkRangeSlider
- ctkTreeComboBox

## Expected behavior

All widgets that are available in Qt Designer should be usable with the parameter node wrapper's GUI connector.

## Environment
- Slicer version: Slicer-5.5.0-2023-10-26
- Operating system: all

**Comment by HarryDC** [comment]
I think we can turn this into a tracking issue for the unsupported widgets but there is a difference in the amount of effort it will take to implement some of these. For the following we already support the underlying data types which means we only need to implement one side of the mapping (from the widget to the datatype) 

- [ ] qMRMLListWidget
- [ ] qMRMLRangeWidget
- [ ] qMRMLSpinBox
- [ ] ctkCheckBox
- [ ] ctkComboBox
- [ ] ctkDoubleRangeSlider
- [ ] ctkDoubleSlider
- [ ] ctkDoubleSpinBox
- [ ] ctkRangeSlider

these might work but we'll have to do some triage 

- [ ] qMRMLCheckableNodeComboBox
- [ ] ctkCheckableComboBox
- [ ] ctkFittedTextBrowser

and these will also need support for the datastructures that they map to

- [ ] qMRMLColorPickerWidget
- [ ] qMRMLCoordinatesWidget
- [ ] qMRMLLinearTransformSlider
- [ ] qMRMLMatrixWidget
- [ ] qMRMLTransformSliders
- [ ] ctkColorPickerButton
- [ ] ctkCoordinatesWidget
- [ ] ctkMatrixWidget
- [ ] ctkTreeComboBox

**Comment by lassoan** [comment]
@HarryDC thanks a lot for looking into this. Do you think you will have time to work on some of these?

**Comment by HarryDC** [comment]
@lassoan Yes I'll start with the "easy" list and we'll work our way down from there but i think PRs are probably going to come in pieces

**Comment by jcfr** [comment]
> re: ctkDoubleRangeSlider

The following pull request should expose the relevant method to Python:

* https://github.com/commontk/CTK/pull/1158

**Comment by HarryDC** [comment]
I can check this but i worked around it using the maximum and minimum property it made sense after the explanation

**Comment by jcfr** [comment]
Using the `setRange` will ensure the range changed signal is emitted only once.

**Comment by HarryDC** [comment]
Note after discussion with @sjh26 we figured out that `qMRMLListWidget` is currently not used in the slicer core, would need more work wrt _what_ the expected result of a selection is for this widget, might need more work in slicer to expose signals on selection.
